File: Python3/1025.py
import bisect

n, q = [int(x) for x in input().split()]
cont = 1

while (n, q) != (0, 0):
    marbles = sorted(int(input()) for _ in range(n))

    print("CASE# {}:".format(cont))
    for j in range(q):
        inp = int(input())
        idx = bisect.bisect_left(marbles, inp)
        if idx < n and marbles[idx] == inp:
            print("{} found at {}".format(inp, idx + 1))
        else:
            print("{} not found".format(inp))
    cont += 1
    n, q = [int(x) for x in input().split()]

File: C++/1025.cpp
#include <bits/stdc++.h>

using namespace std;

int n, q, p, cont, o;
map<int, int> marbles;
map<int, int>::iterator it;

int main(){
	scanf("%d %d", &n, &q);
	while (n != 0 && q != 0){
		marbles.clear();
		
		for (int i = 0; i < n; i++){
			scanf("%d", &p);
			if (marbles.find(p) != marbles.end()){
				marbles[p] += 1;
			} else {
				marbles[p] = 1;
			}
		}

		printf("CASE# %d:\n", ++cont);
		for (int j = 0; j < q; j++){
			scanf("%d", &p);
			o = 0;
			if (marbles.find(p) != marbles.end()){
				
				for (it=marbles.begin(); it != marbles.find(p); it++){
					o += marbles[it->first];
				}
				o++;
				printf("%d found at %d\n", p, o);
			} else {
				printf("%d not found\n", p);
			}
		}
		scanf("%d %d", &n, &q);
	}
	return 0;
}
